Content.Client/CrewMedal/UI/CrewMedalWindow.xaml.cs

```diff
@@ -0,0 +1,66 @@
+using Robust.Client.UserInterface.CustomControls;
+using Robust.Client.AutoGenerated;
+using Robust.Client.UserInterface.XAML;
+
+namespace Content.Client.CrewMedal.UI;
+
+[GenerateTypedNameReferences]
+public sealed partial class CrewMedalWindow : DefaultWindow
+{
+    public event Action<string>? OnReasonChanged;
+
+    /// <summary>
+    /// Is the user currently entering text into the control?
+    /// </summary>
+    private bool _focused;
+
+    private string _reason = string.Empty;
+    private bool _awarded = false;
+
+    private int _maxCharacters = 50;
+
+    public CrewMedalWindow()
+    {
+        RobustXamlLoader.Load(this);
+
+        ReasonLineEdit.OnTextChanged += _ =>
+        {
+            SaveButton.Disabled = _awarded || ReasonLineEdit.Text.Length > _maxCharacters;
```

Comment:
Appreciate that the first part of this is to re-enable the button after a save, but for the second condition, I think it would be better to use LineEdit.IsValid to prevent users from typing in too much text in the first place.